More fixes related to #153

PaulSD opened the pull request:
After more testing, I found and fixed a bunch of issues with the commits in #153.
To avoid having broken commits in the history (in case anyone needs to bisect bugs), I've corrected the commits instead of adding more commits to fix them.

@ildar170975 Please discard the current `dev` branch and re-create it with these commits.

Built copy for anyone who wants to test this: https://github.com/PaulSD/config-template-card/releases/tag/1.3.6.2

swiergot committed:
Allow multiple templates anywhere in property value

swiergot committed:
Fix for non-string properties

swiergot committed:
Merge branch 'master' into improved_eval

malmeloo committed:
Update dependencies + cleanup

PaulSD committed:
Allow population of the entire `entities` list using templates

Fixes #93, Closes #119, Closes #147
(This eliminates the need for #119 and #147)

PaulSD committed:
Define named variables as variables are parsed

If multiple variables are defined then subsequent variables can
currently reference prior variables using `vars['name']` but not just
`name`.  This enables variables to reference prior variables by name,
like templates can.

PaulSD committed:
Combine and simplify _evaluateConfig() and _evaluateArray()

PaulSD committed:
README.md: Update function example to show the use of multiple args

PaulSD committed:
Update dependencies and clean up

Use lit 2.8.0 to match HA frontend version.

Fixes #91, Closes #123, Closes #128, Closes #129, Closes #135

ildar170975 committed:
version fix

ildar170975 commented:
> Please discard the current dev branch and re-create it with these commits.

Do you mean:
-- just delete the "dev" branch,
-- re-create dev based on "base",
-- merge THIS pull request into "dev" - and this will contain ALL commits (old unreleased + you required fixes)
?

PaulSD commented:
Recreate the dev branch from master, then pull these changes into dev.  Discard the old unreleased commits.

ildar170975 commented:
>  Discard the old unreleased commits.

Why it is needed? These commits do work (like from https://github.com/iantrich/config-template-card/pull/109).
Let's just add your changes?

PaulSD commented:
I was trying to make the commit history more sane.  But, OK, we can just add my changes.
Let me rework this PR...

ildar170975 commented:
May be I understood you in a wrong way.
I wanted the history keep actual authors (all changes added by John Smith should be marked as "of John Smith").

ildar170975 commented:
Also, could you take a look at ALL workflow files? Imho they need to be updated.

PaulSD commented:
Ok, will look.

PaulSD committed:
Replace old deepClone() with modern structuredClone()

PaulSD committed:
Don't throw confusing exceptions if there are incomplete templates

(Fix for 32da89f3)

PaulSD committed:
Simplify code using ?. and += notation

PaulSD committed:
Merge pull request #110 from swiergot/improved_eval

Allow multiple templates anywhere in property value

PaulSD committed:
Merge pull request #148 from malmeloo:chore/cleanup

PaulSD committed:
eval() variables once per render instead of for every template

This will drastically improve performance if any variable templates are
expensive to eval.

PaulSD commented:
Ok, I've fixed the authors, updated the workflow files, fixed a few more issues in the code.

These commits don't apply cleanly on top of my prior commits in the dev branch, so we still need to re-create `dev` from `master` then apply this PR on `dev`.

ildar170975 commented:
@PaulSD 
Thanks a lot! Let's merge it to recreated dev.

ildar170975 commented:
@PaulSD 
Can we add a js file to "assets"? Cannot find it.
